tools/clang/lib/SPIRV/SPIRVEmitter.cpp

```diff
@@ -2166,9 +2215,18 @@ uint32_t SPIRVEmitter::processIntrinsicAsType(const CallExpr *callExpr) {
     return doExpr(arg);
 
   if (hlsl::IsHLSLMatType(argType)) {
-    emitError("'asfloat', 'asint', and 'asuint' do not support matrix "
-              "arguments yet.");
-    return 0;
+    uint32_t rowCount = 0, colCount = 0;
```

Comment:
Minor: can use `isSpirvAcceptableMatType` here directly to hide the two `if` checks and size fetching call.